### PR TITLE
Cli and gem

### PR DESCRIPTION
**Pull request by davepagurek**
Ok MASSIVE DIFF TIME, here is what's changed:

- dependencies are now in `emerald.gemspec`
- exported binaries (`emerald` in this case) are also listed in the gemspec
- classes should be namespaced under the Emerald module now, although I haven't moved them all yet (just Preprocessor because it's being tested)
- tests use rspec now (no change in terms of running them, still `rake test`, but we get nice output for free, and any `*_spec` files now get run automatically in CI without us changing anything)
- you can run `bin/emerald process [file] [-b]` to convert to html

### CODE REVIEW

**Comment by armcburney** [comment]
`71 files changed` - checks out

File: LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2016 Dave Pagurek
```
Comment (armcburney) [inline LICENSE.txt]: Can you add Andrew McBurney here too please 😃 (and Yu Chen too)

Comment (armcburney) [reply LICENSE.txt]: If he wants to be included

File: emerald.gemspec [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# coding: utf-8
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'emerald/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'emerald'
+  spec.version       = Emerald::VERSION
+  spec.authors       = ['Andrew McBurney']
```
Comment (armcburney) [inline emerald.gemspec]: Dave, add yo-self

File: lib/emerald/PreProcessor.rb [hunk 3]
```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env ruby
```
Comment (armcburney) [inline lib/emerald/PreProcessor.rb:1]: Was there anything changed to this file, or does git just not recognize the file has been moved?

File: lib/emerald/version.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module Emerald
+  VERSION = "0.1.0"
```
Comment (armcburney) [inline lib/emerald/version.rb:2]: Wow, we're like - legit now lol

File: spec/functional/scope_spec.rb [hunk 5]
```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+SCOPE_TESTS = [
+
```
Comment (armcburney) [inline spec/functional/scope_spec.rb]: hey, do we need this extra whitespace here?

File: spec/functional/scope_spec.rb [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+ 
+  
+].freeze
+
+#
+# Unit testing for Emerald Language. Asserts that emerald passed in is
+# compiled to valid html, which is equivalent in semantic value.
+#
+describe Emerald do
+
```
Comment (armcburney) [inline spec/functional/scope_spec.rb]: same thing here, do we need this extra whitespace?

File: spec/functional/scope_spec.rb [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    it 'can determine truthiness' do
+      expect(convert(
+        context: {test: true},
+        input: <<~EMR,
+          given test
+            h1 True
+          unless test
+            h1 False
+        EMR
+      )).to eq('<h1>True</h1>')
```
Comment (armcburney) [inline spec/functional/scope_spec.rb:21]: nice, I like this a lot better than the previous tests. Great job changing all these over to rspec!

**Comment by armcburney** [comment]
You put the team on your back with this PR Dave, great job with everything!! Just a few comments and then I'll approve. Awesome job!

File: spec/functional/scope_spec.rb (hunk 5 above)
Comment (davepagurek) [reply spec/functional/scope_spec.rb]: Oh this is an empty array, I can just remove that haha

File: lib/emerald/PreProcessor.rb (hunk 3 above)
Comment (davepagurek) [reply lib/emerald/PreProcessor.rb:1]: I put it under the Emerald module namespace and that's it

**Review by armcburney — APPROVED** [review]
Great job Dave, I would give you a cookie, but you're in cali and I'm in the 6ix, so this will have to do: 🍪

**Pull request merged**